modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheContextInfo.java

```diff
@@ -62,20 +67,22 @@ public GridCacheContextInfo(GridCacheContext<K, V> cctx, boolean clientCache) {
         this.clientCache = clientCache;
 
         this.cctx = cctx;
+        this.kctx = kctx;
     }
 
     /**
      * Constructor of not started cache context.
      *
      * @param cacheDesc Cache descriptor.
      */
-    public GridCacheContextInfo(DynamicCacheDescriptor cacheDesc) {
+    public GridCacheContextInfo(DynamicCacheDescriptor cacheDesc, GridKernalContext ctx) {
         config = cacheDesc.cacheConfiguration();
         dynamicDeploymentId = cacheDesc.deploymentId();
         groupId = cacheDesc.groupId();
         cacheId = CU.cacheId(config.getName());
 
         clientCache = true;
+        this.kctx = ctx;
```

Comment:
You don't need to pass `KernalContext` here.
`KernalContext` can be get from `DynamicCacheDescriptor`,  there is a unused parameter in  `DynamicCacheDescriptor` constructor.